core/src/main/java/com/yahoo/tagchowder/Parser.java

```diff
@@ -1185,6 +1204,27 @@ private String makeName(final char[] buff, final int offset, final int length) {
                 }
                 start = true;
                 dst.append(translateColons ? '_' : ch);
+            } else if (ampValidation) {
+                //if the beginning character starts with '[' or '{', ensure the brackets are balanced.
+                //allows [text], [[text]], {text}, {{text}}..etc
+                if (start && (ch == '[' || ch == '{'))  {
+                    if (areBracketsBalanced(buff, offset, length)) {
+                        dst.append(ch);
+                        bracketsAreBalanced = true;
+                    }
+                } else if (start && ch == '⚡') {
+                    //if the beginning character starts with '⚡', ensure that this buffer with offset and length contains "⚡4email"
+                    final StringBuffer str = new StringBuffer(length);
```

Comment:
Talked to Luis, he does not want any "new" memory allocation, can you fix the code?